### PR TITLE
Feature/add editing

### PR DESCRIPTION
**Pull request by jonathanstallings**
Permalinks, editing, markdown, and code highlighting have been added.

### CODE REVIEW

File: journal.py [hunk 1]
```diff
@@ -64,12 +65,27 @@ def write(cls, title=None, text=None, session=None):
...
 
 
+def apply_markdown(text):
+    return markdown.markdown(
+        text,
+        extensions=['codehilite', 'fenced_code'],
+        ouput_format="html5"
+    )
+
+
```
Comment (cewing) [inline journal.py:88]: If this is a method on the entry class that gets `self` as its first argument, then you can simply call `entry.apply_markdown()`, right?  That'd be a bunch cleaner.

File: templates/detail.jinja2 [hunk 2]
```diff
@@ -6,10 +6,10 @@
         <h1>{{ entry.title|safe }}</h1>
         <span class="date">{{ entry.created.strftime('%b. %d, %Y') }}</span>
         {% if request.authenticated_userid %}
-        <a class="button pull-right" href="{{ request.route_url('edit', id=entry.id) }}">Edit</a>
+        <a id="edit-button" class="button pull-right" href="{{ request.route_url('edit', id=entry.id) }}">Edit</a>
         {% endif %}
         <hr />
-        <div>{{ entry.text|safe }}</div>
+        <div id="body-text">{{ apply_markdown(entry.text)|safe }}</div>
```
Comment (cewing) [inline templates/detail.jinja2:12]:
``` jinja
{{ entry.apply_markdown()|safe }}
```

See?  prettier.

File: journal.py (hunk 1 above)
Comment (jonathanstallings) [inline journal.py:88]: Agreed!

File: templates/detail.jinja2 (hunk 2 above)
Comment (jonathanstallings) [inline templates/detail.jinja2:12]: Much!

**Pull request merged**